src/main/webapp/app/components/tokenInputGroups/TokenExtendButton.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import {
+  InputGroup,
+  Button,
+  Dropdown,
+  DropdownButton,
+  Row
+} from 'react-bootstrap';
+import classnames from 'classnames';
+import { getMomentInstance } from 'app/shared/utils/Utils';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import moment from 'moment';
+import {
+  APP_LOCAL_DATETIME_FORMAT_Z_FORCE,
+  APP_LOCAL_DATE_FORMAT
+} from 'app/config/constants';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+
+type TokenExtendButtonProps = {
+  currentDate: string;
+  afterChangeDate: (newDate: string) => void;
+};
+
+export const TokenExtendButton: React.FunctionComponent<TokenExtendButtonProps> = props => {
+  let selectedDay: Date;
+
+  function handleDayChange(day: any) {
+    selectedDay = day;
+  }
+
+  function handleSelectedDay() {
+    props.afterChangeDate(
+      moment(selectedDay).format(APP_LOCAL_DATETIME_FORMAT_Z_FORCE)
+    );
+  }
+
+  const extendSchedule = [
+    {
+      amount: 1,
+      scheduleIndex: 1,
+      unitText: 'week'
+    },
+    {
+      amount: 1,
+      scheduleIndex: 2,
+      unitText: 'month'
+    },
+    {
+      amount: 6,
+      scheduleIndex: 3,
+      unitText: 'months'
+    },
+    {
+      amount: 1,
+      scheduleIndex: 4,
+      unitText: 'year'
+    }
+  ];
+
+  return (
+    <DefaultTooltip placement={'top'} overlay={'Change token expiration date'}>
+      <DropdownButton
+        as={InputGroup.Append}
+        title={<i className={classnames('fa fa-calendar')}></i>}
+        id="time-entend-dropdown"
+      >
+        {extendSchedule.map(val => {
+          let unit: moment.unitOfTime.DurationConstructor;
+          switch (val.unitText) {
+            case 'day':
+            case 'days':
+              unit = 'd';
+              break;
+            case 'week':
+            case 'weeks':
+              unit = 'w';
+              break;
+            case 'month':
+            case 'months':
+              unit = 'M';
+              break;
+            case 'year':
+            case 'years':
+              unit = 'y';
+              break;
+            default:
+              unit = 'd';
+              break;
+          }
```

Comment:
Again, I think you just need to include unit in the extendSchedule definition. No need to add a switch here.